Enterprise-CMCS/macpro-onemac
# OY2 24303: TE Request should pre-fill TE Type

Story: https://qmacbis.atlassian.net/browse/OY2-24303
Endpoint: See github-actions bot comment

### Details
Since the TE type depends on the Waiver package being extended... makes sense that when starting a Temporary Extension Request from an existing package, we pre fill the Type. Type is associated with the Waiver Authority, but is at a higher level, so 1915(b) and 1915(b)(4) are both 1915(b)s for the purpose of Extensions.

### Changes
- added temporaryExtensionType as a var passed in the FormLocationState
- getMyPackages and getDetails now return temporaryExtensionType as part of the package item
- modified TemporaryExtensionTypeInput to accept a pre fill and "disable" the component in the same way we do the ComponentIds

### Implementation Notes
- Did use the fact that 1915(b) and 1915(b)(4) both start with 1915(b) to set the value for Temp Extension Type
- may want to move this to package build time? 

### Test Plan
1. Login as a submitting user
2. Navigate to Waivers tab on Dashboard
3. Find an Initial Waiver or Waiver Renewal in "Approved" status
4. click the Action dots, verify the type is prefilled
5. Verify can submit the Temporary Extension
6. Return to Dashboard
7. Click into the package details of an appropriate package
8. Click the "Request Temporary Extension" link in Package Actions
9. Verify the type is prefilled
10. Verify you can create the extension
11. use the New Submission button 
12. Triage to Temporary Extension Form
13. Verify the blank version of form still works as expected.
